### ISSUE TITLE
extending HTMLElement not working for variants (nav, footer)

### ISSUE THREAD

**Issue by domtra** [WebReflection/custom-elements-builtin#2]
Is it expected, that it is not possible to extend a nav or footer element (both an HTMLElement), for example, or is it a bug?

This is working with document-register-element. The versions of custom-element that I tried all did not work.

Here is a [codepen](https://codepen.io/domtra/pen/VwjbvVM) to reproduce the issue. If you remove the polyfill, it will work in Chrome, for example. Error messages are

Chrome: TypeError: Illegal constructor
Safari: TypeError: new.target does not define a custom element

**Comment by WebReflection** [comment]
You should **not** use this polyfill in Chrome but I had a similar issue in [Life Diary](https://github.com/WebReflection/life-diary) and WebKit, where the footer works without issues thanks to [uce-template](https://github.com/WebReflection/uce-template#readme) taking over.

It might be that the logic needs some improvement though, so thanks for filing this bug, yet if you need a polyfill please use [@ungap/custom-elements](https://github.com/ungap/custom-elements) because this polyfill has no features detection, while the full polyfill does.

**Comment by domtra** [comment]
I was not going to use it for chrome. I just thought it might be easier to debug there than in safari because of the dev tools, and that the underlying issue would be the same.
I was going to load the polyfill conditionally based on the code example in the readme. But safari actually does not throw an error when using `extends`. Maybe this should be updated to the logic you use in `@ungap/custom-elements`.

And just for clarification: `@ungap/custom-elements` will always be the 'official' / stable release of the polyfill? I am just a bit confused, because the version does not match the one from `@webreflection/custom-elements`. It seems to be the other way around for `custom-elements-builtin`. `@ungap` has a higher version but older code than `@webreflection`. But those should not be used directly?

**Comment by WebReflection** [comment]
the `@ungap/custom-elements` versioning doesn't reflect its underlying modules, but it uses the very latest behind the scene. These polyfills can be used as stand-alone in specific cases (i.e. some WPEWebKit project that uses custom elements) but `@ungap/custom-elements` has the right features detection to patch everything down to IE11, patch only builtin extends in Safari/WebKit, and patch absolutely nothing in Firefox and Chrome.

The "*dance*" to properly feature detect everything is there, and if WebKit/Safari changes (again) something, it's in there that will be patched, not in this README (although I should remember to eventually update the feature detection in here too).

So, long story short, *yes*, `@ungap/custom-elements` *is* the official polyfill, and it gets updated as soon as any of these two change.

**Comment by domtra** [comment]
thanks for the info @WebReflection!

**Comment by WebReflection** [comment]
As example ... I've now broken everything but `@ungap/custom-elements` is still working ... it will be updated once I manage to properly fix this issue

**Comment by domtra** [comment]
For me it seems to be working with your changes. In the codepen and also in a local project.

**Comment by WebReflection** [comment]
> For me it seems to be working with your changes. In the codepen and also in a local project.

it breaks on regular custom elements now ... which is not good

**Comment by WebReflection** [comment]
reverted (it's back as it was 'cause I haven't found a proper fix yet)

**Comment by WebReflection** [comment]
Everything has been fixed/updated and it should work out of the box down to IE11.
